gsy_framework/serialization.py

```diff
@@ -0,0 +1,17 @@
+import pickle
+from typing import Dict
+from zlib import compress, decompress
+
+
+class CompressSerialization:
```

Comment:
I think we can find a way to further improve this name. On one side, we try to use nouns as class names and not verbs, so in this case it could be something like `SerializationCompressor`. But I would still not be completely convinced.
This class does: 
- compression/de-compression
- encoding/decoding
- Also, in the docstring you mention that this is a serializer/deserializer. 

What about just calling it something like `DataSerializer`? There are probably better alternatives, so feel free to suggest any other option.